posthog/tasks/test/test_email.py

```diff
@@ -1,39 +1,137 @@
-from typing import Any
+from typing import Any, Tuple
 from unittest.mock import MagicMock, patch
 
 from freezegun import freeze_time
 
 from posthog.models import Organization, Team, User
 from posthog.models.instance_setting import set_instance_setting
 from posthog.models.messaging import MessagingRecord, get_email_hash
-from posthog.models.organization import OrganizationMembership
-from posthog.tasks.email import send_first_ingestion_reminder_emails, send_second_ingestion_reminder_emails
+from posthog.models.organization import OrganizationInvite, OrganizationMembership
+from posthog.models.plugin import Plugin, PluginConfig
+from posthog.tasks.email import (
+    send_async_migration_complete_email,
+    send_async_migration_errored_email,
+    send_canary_email,
+    send_fatal_plugin_error,
+    send_first_ingestion_reminder_emails,
+    send_invite,
+    send_member_join,
+    send_password_reset,
+    send_second_ingestion_reminder_emails,
+)
 from posthog.tasks.test.utils_email_tests import mock_email_messages
 from posthog.test.base import APIBaseTest, ClickhouseTestMixin
 
 
-def create_org_team_and_user(creation_date: str, email: str, ingested_event: bool = False) -> Organization:
+def create_org_team_and_user(creation_date: str, email: str, ingested_event: bool = False) -> Tuple[Organization, User]:
     with freeze_time(creation_date):
         org = Organization.objects.create(name="too_late_org")
         Team.objects.create(organization=org, name="Default Project", ingested_event=ingested_event)
-        User.objects.create_and_join(
+        user = User.objects.create_and_join(
             organization=org, email=email, password=None, level=OrganizationMembership.Level.OWNER,
         )
-        return org
+        return org, user
 
 
 @patch("posthog.tasks.email.EmailMessage")
 class TestEmail(APIBaseTest, ClickhouseTestMixin):
-    @patch("posthoganalytics.feature_enabled", return_value=True)
-    def test_first_email_sent_to_correct_users_only_once(self, __: Any, MockEmailMessage: MagicMock) -> None:
-        mocked_email_messages = mock_email_messages(MockEmailMessage)
+    """
+        NOTE: Every task in the "email" tasks should have at least one test.
+        using the `mock_email_messages` helper writes the email output to `tasks/test/__emails__`
+        so you can check out what it is rendered 🙌
```

Comment:
Is it too wild to store a golden master for each email file and then compare the rendered files?

If they hash to the same value or some similar comparison?

Not blocking since it's awesome we have the tests that are here just 🤔

Reply:
I considered that. My "concern" at the time was things like IDs or timestamps that are different but that concern might be unwarranted. I'll perhaps lean towards merging this but keep that in mind as a follow up addition